report/renderer/report-ui-features.js

```diff
@@ -322,20 +322,33 @@ export class ReportUIFeatures {
    * @return {Array<HTMLElement>}
    */
   _getThirdPartyRows(rowEls, finalDisplayedUrl) {
-    /** @type {Array<HTMLElement>} */
-    const thirdPartyRows = [];
     const finalDisplayedUrlRootDomain = Util.getRootDomain(finalDisplayedUrl);
 
-    for (const rowEl of rowEls) {
-      if (rowEl.classList.contains('lh-sub-item-row')) continue;
-
-      const urlItem = rowEl.querySelector('div.lh-text__url');
-      if (!urlItem) continue;
+    /** @type {string | undefined} */
+    let mainEntityName;
+    if (this.json.audits['entity-classification'] &&
+        this.json.audits['entity-classification'].details &&
+        this.json.audits['entity-classification'].details.type === 'entity-classification') {
+      /** @type {LH.Audit.Details.EntityClassification} */
+      const entityClassification = this.json.audits['entity-classification'].details;
+      mainEntityName = entityClassification.firstParty;
```

Comment:
The computed artifact implies this may be undefined if `data.URL?.mainDocumentUrl || data.URL?.finalDisplayedUrl` is falsy. Is that ever true? If it isn't, can you add a comment somewhere by that code or on the type

Reply:
`.firstParty` could be empty, when a first party cannot be accurately determined (either due to missing URLs, like timespans or snapshot). 

As for the `entity-classification` itself missing (if thats what you were referring to), it's possible in older LHRs.